### DISCUSSION TITLE
Error after I try : python -m pip install -r requirements.txt on windows 10

### DISCUSSION THREAD

**Discussion by phearakthefoundry** [saleor/saleor#6446] [Q&A]
Running setup.py install for pillow ... error
    ERROR: Command errored out with exit status 1:
     command: 'C:\Users\PC\Desktop\ManualSaleor\saleor\saleor-venv\Scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\PC\\AppData\\Local\\Temp\\pip-install-g3w_0z9l\\pillow\\setup.py'"'"'; __file__='"'"'C:\\Users\\PC\\AppData\\Local\\Temp\\pip-install-g3w_0z9l\\pillow\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\PC\AppData\Local\Temp\pip-record-ita8dway\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\PC\Desktop\ManualSaleor\saleor\saleor-venv\include\site\python3.9\pillow'
         cwd: C:\Users\PC\AppData\Local\Temp\pip-install-g3w_0z9l\pillow\
    Complete output (174 lines):
    C:\Users\PC\AppData\Local\Temp\pip-install-g3w_0z9l\pillow\setup.py:42: RuntimeWarning: Pillow 7.2.0 does not support Python 3.9 and does not provide prebuilt Windows binaries. We do not recommend building from source on Windows.
      warnings.warn(
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.9
    creating build\lib.win-amd64-3.9\PIL
    copying src\PIL\BdfFontFile.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\BlpImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\BmpImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\BufrStubImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ContainerIO.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\CurImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\DcxImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\DdsImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\EpsImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ExifTags.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\features.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\FitsStubImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\FliImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\FontFile.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\FpxImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\FtexImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\GbrImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\GdImageFile.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\GifImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\GimpGradientFile.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\GimpPaletteFile.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\GribStubImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\Hdf5StubImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\IcnsImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\IcoImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\Image.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageChops.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageCms.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageColor.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageDraw.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageDraw2.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageEnhance.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageFile.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageFilter.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageFont.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageGrab.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageMath.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageMode.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageMorph.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageOps.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImagePalette.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImagePath.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageQt.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageSequence.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageShow.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageStat.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageTk.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageTransform.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImageWin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\ImtImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\IptcImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\Jpeg2KImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\JpegImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\JpegPresets.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\McIdasImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\MicImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\MpegImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\MpoImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\MspImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\PaletteFile.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\PalmImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\PcdImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\PcfFontFile.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\PcxImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\PdfImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\PdfParser.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\PixarImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\PngImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\PpmImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\PsdImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\PSDraw.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\PyAccess.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\SgiImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\SpiderImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\SunImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\TarIO.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\TgaImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\TiffImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\TiffTags.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\WalImageFile.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\WebPImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\WmfImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\XbmImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\XpmImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\XVThumbImagePlugin.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\_binary.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\_tkinter_finder.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\_util.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\_version.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\__init__.py -> build\lib.win-amd64-3.9\PIL
    copying src\PIL\__main__.py -> build\lib.win-amd64-3.9\PIL
    running egg_info
    writing src\Pillow.egg-info\PKG-INFO
    writing dependency_links to src\Pillow.egg-info\dependency_links.txt
    writing top-level names to src\Pillow.egg-info\top_level.txt
    reading manifest file 'src\Pillow.egg-info\SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no files found matching '*.c'
    warning: no files found matching '*.h'
    warning: no files found matching '*.sh'
    warning: no previously-included files found matching '.appveyor.yml'
    warning: no previously-included files found matching '.coveragerc'
    warning: no previously-included files found matching '.editorconfig'
    warning: no previously-included files found matching '.readthedocs.yml'
    warning: no previously-included files found matching 'codecov.yml'
    warning: no previously-included files matching '.git*' found anywhere in distribution
    warning: no previously-included files matching '*.pyc' found anywhere in distribution
    warning: no previously-included files matching '*.so' found anywhere in distribution
    no previously-included directories found matching '.ci'
    writing manifest file 'src\Pillow.egg-info\SOURCES.txt'
    running build_ext

    The headers or library files could not be found for zlib,
    a required dependency when compiling Pillow from source.

    Please see the install instructions at:
       https://pillow.readthedocs.io/en/latest/installation.html

    Traceback (most recent call last):
      File "C:\Users\PC\AppData\Local\Temp\pip-install-g3w_0z9l\pillow\setup.py", line 864, in <module>
        setup(
      File "C:\Users\PC\Desktop\ManualSaleor\saleor\saleor-venv\lib\site-packages\setuptools\__init__.py", line 165, in setup
        return distutils.core.setup(**attrs)
      File "C:\Users\PC\AppData\Local\Programs\Python\Python39\lib\distutils\core.py", line 148, in setup
        dist.run_commands()
      File "C:\Users\PC\AppData\Local\Programs\Python\Python39\lib\distutils\dist.py", line 966, in run_commands
        self.run_command(cmd)
      File "C:\Users\PC\AppData\Local\Programs\Python\Python39\lib\distutils\dist.py", line 985, in run_command
        cmd_obj.run()
      File "C:\Users\PC\Desktop\ManualSaleor\saleor\saleor-venv\lib\site-packages\setuptools\command\install.py", line 61, in run
        return orig.install.run(self)
      File "C:\Users\PC\AppData\Local\Programs\Python\Python39\lib\distutils\command\install.py", line 546, in run
        self.run_command('build')
      File "C:\Users\PC\AppData\Local\Programs\Python\Python39\lib\distutils\cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "C:\Users\PC\AppData\Local\Programs\Python\Python39\lib\distutils\dist.py", line 985, in run_command
        cmd_obj.run()
      File "C:\Users\PC\AppData\Local\Programs\Python\Python39\lib\distutils\command\build.py", line 135, in run
        self.run_command(cmd_name)
      File "C:\Users\PC\AppData\Local\Programs\Python\Python39\lib\distutils\cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "C:\Users\PC\AppData\Local\Programs\Python\Python39\lib\distutils\dist.py", line 985, in run_command
        cmd_obj.run()
      File "C:\Users\PC\AppData\Local\Programs\Python\Python39\lib\distutils\command\build_ext.py", line 340, in run
        self.build_extensions()
      File "C:\Users\PC\AppData\Local\Temp\pip-install-g3w_0z9l\pillow\setup.py", line 694, in build_extensions
        raise RequiredDependencyException(f)
    __main__.RequiredDependencyException: zlib

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\PC\AppData\Local\Temp\pip-install-g3w_0z9l\pillow\setup.py", line 918, in <module>
        raise RequiredDependencyException(msg)
    __main__.RequiredDependencyException:

    The headers or library files could not be found for zlib,
    a required dependency when compiling Pillow from source.

    Please see the install instructions at:
       https://pillow.readthedocs.io/en/latest/installation.html

    ----------------------------------------
  Rolling back uninstall of pillow
  Moving to c:\users\pc\desktop\manualsaleor\saleor\saleor-venv\lib\site-packages\pillow-8.0.1-py3.9-win-amd64.egg
   from C:\Users\PC\Desktop\ManualSaleor\saleor\saleor-venv\Lib\site-packages\~illow-8.0.1-py3.9-win-amd64.egg
ERROR: Command errored out with exit status 1: 'C:\Users\PC\Desktop\ManualSaleor\saleor\saleor-venv\Scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\PC\\AppData\\Local\\Temp\\pip-install-g3w_0z9l\\pillow\\setup.py'"'"'; __file__='"'"'C:\\Users\\PC\\AppData\\Local\\Temp\\pip-install-g3w_0z9l\\pillow\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\PC\AppData\Local\Temp\pip-record-ita8dway\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\PC\Desktop\ManualSaleor\saleor\saleor-venv\include\site\python3.9\pillow' Check the logs for full command output.
WARNING: You are using pip version 20.2.3; however, version 20.2.4 is available.
You should consider upgrading via the 'C:\Users\PC\Desktop\ManualSaleor\saleor\saleor-venv\Scripts\python.exe -m pip install --upgrade pip' command.

**Comment by maarcingebala** [comment]
Pillow requires a few system dependencies (in this case `zlib`) which are missing on your computer and it fails to install. I'm not familiar with solving this kind of issues on Windows, but here is a related thread which may help you: https://github.com/python-pillow/Pillow/issues/4242

**Reply by phearakthefoundry** [reply]
the problem still has and I stopped install manual on windows, I started to use saleor-platform with docker and it work find but currently I would like to know that How can I deploy saleor-platform as production mode? cuz for now I have customized it for deployment.

**Reply by karolkielecki** [reply]
@phearakthefoundry Hey, I just replied to you here: https://github.com/mirumee/saleor/discussions/6453#discussioncomment-127808
